### PR TITLE
feat: register personal account api request (AR-953)

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

AR-953

### Solutions

implementing register account and activation BE requests

Needs releases with:

- [ ] GitHub link to other pull request

### Testing

unit test for `RegisterApi`

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: network/src/commonTest/kotlin/com/wire/kalium/api/ApiTest.kt [hunk 1]
```diff
@@ -165,7 +166,13 @@ interface ApiTest {
     // body
     fun HttpRequestData.assertBodyContent(content: String) {
         assertIs<TextContent>(body)
-        assertEquals(content, (body as TextContent).text)
+        // convert both body and the content to JsonObject, so we are not comparing strings
+        // since json strings can have different orders of values
+        val expected = buildJsonObject { buildString { content } }
+        println(expected)
```
Comment (typfel) [inline network/src/commonTest/kotlin/com/wire/kalium/api/ApiTest.kt:172]: debug code

File: network/src/commonTest/kotlin/com/wire/kalium/api/ApiTest.kt [hunk 2]
```diff
@@ -165,7 +166,13 @@ interface ApiTest {
...
+        val actual = buildJsonObject { buildString { (body as TextContent).text } }
+        println(actual)
```
Comment (typfel) [inline network/src/commonTest/kotlin/com/wire/kalium/api/ApiTest.kt:174]: debug code

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/MessageSender.kt [hunk 3]
```diff
@@ -53,20 +52,19 @@ class MessageSenderImpl(
...
-
-        if (sendResult.isRight()) {
-            return@suspending Either.Right(Unit)
-        }
-
-        when (val failure = sendResult.value) {
-            is SendMessageFailure.ClientsHaveChanged -> {
-                messageSendFailureHandler.handleClientsHaveChangedFailure(failure).flatMap {
-                    getRecipientsAndAttemptSend(conversationId, messageUuid)
+        messageRepository.sendEnvelope(conversationId, envelope).coFold(
```
Comment (typfel) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/MessageSender.kt:55]: Unrelated changes?

**Review by typfel — CHANGES_REQUESTED** [review]
👍

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/MessageSender.kt (hunk 3 above)
Comment (MohamadJaara) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/MessageSender.kt]: this is for a different PR i will revert the changes

**Review by typfel — APPROVED** [review]
